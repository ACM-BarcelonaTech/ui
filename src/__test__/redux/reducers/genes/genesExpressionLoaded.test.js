import genesExpressionLoadedReducer from 'redux/reducers/genes/genesExpressionLoaded';

describe('genesExpressionLoaded reducer', () => {
  it('returns correct state', () => {
    const initialState = {
      expression: {
        views: {},
        data: {},
      },
      markers: {},
    };

    const genes = ['geneA'];
    const data = {
      geneA: {
        rawExpression: {
          expression: [1],
          mean: 1,
          stdev: 1,
        },
      },
    };

    const newState = genesExpressionLoadedReducer(initialState, {
      payload: {
        genes,
        data,
        plotUuid: 'interactiveHeatmap',
      },
    });

    expect(newState).toMatchSnapshot();
  });
});
