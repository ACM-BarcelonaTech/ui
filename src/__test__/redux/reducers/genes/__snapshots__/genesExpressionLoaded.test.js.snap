// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genesExpressionLoaded reducer returns correct state 1`] = `
Object {
  "expression": Object {
    "data": Object {
      "geneA": Object {
        "rawExpression": Object {
          "expression": Array [
            1,
          ],
          "mean": 1,
          "stdev": 1,
        },
        "zScore": Array [
          0,
        ],
      },
    },
    "loading": Array [],
    "views": Object {
      "undefined": Object {
        "data": Array [
          "geneA",
        ],
        "error": false,
        "fetching": false,
      },
    },
  },
  "markers": Object {},
}
`;
